src/libos/src/unit_test/mod.rs

```diff
@@ -0,0 +1,118 @@
+mod test;
+
+use super::*;
+use std::panic;
+use std::sync::atomic::{AtomicIsize, Ordering};
+use util::mem_util::*;
+
+static NUM_PASS: AtomicIsize = AtomicIsize::new(0);
+static NUM_FAIL: AtomicIsize = AtomicIsize::new(0);
+
+const GREEN_OK: &str = "\x1b[0;32mok\x1b[0m";
+const RED_FAILED: &str = "\x1b[0;31mFAILED\x1b[0m";
+
+pub fn unit_test(cmd: usize, func_name_ptr: *const c_char, index: i32) -> Result<isize> {
```

Comment:
I think the interface of this `unit_test` syscall can be improved to the following form:
```
pub fn unit_test(test_prefix: *const c_char, int *total_count, int *pass_count) -> Result<isize>
```
with the arguments
* `test_prefix` is an optional input, specifying the prefix of the unit tests should be run. If not given, all test cases should be run.
* `total_count` is an output, giving the total number of test cases that run.
* `pass_count` is an output, giving the number of test cases that pass.

The new interface is not only more powerful (allowing prefix/exact matching), but also simplifies the user and kernel code 
* The user program that drives the unit tests (`test/unit_test/main.c`) can be greatly simplified;
* The user program that runs individual test case (`test/syscall/main.c`) is no longer needed;
* The kernel code does not need to maintain state (e.g., `NUM_PASS` or `NUM_FAIL`) and does not need to have separate commands (`GetCount`, `GetPass`, `RunByName`, and `RunByIndex`);
* Each unit test does need to be run in separate syscall (I cannot see why this is more desirable).

Reply:
I removed `test/syscall`.